testData.js

```diff
@@ -5,11 +5,15 @@ var HTTP = require("./http");
 var Transcript = require("./transcript");
 
 var config = require("./config.json");
+const fs = require("fs");
+const path = require('path');
+
+const readFile = require('util').promisify(fs.readFile);
 
 class TestData {
 
     constructor(obj, query) {
-        this.name = (query && query.name) || (obj && obj.name) || (query && query.url.split('/').pop());
+        this.name = (query && query.name) || (obj && obj.name) || (query?.url?.split('/').pop()) || (query?.path?.split('/').pop());

```

Comment:
I don't like the mutating array functions... in this case it's okay since the array is created on the fly by `split` and never used again, but just so you know, immutable pop:
```
arr.slice(-1)[0]
```